### DISCUSSION TITLE
Add Discussion to Community KB on website

### DISCUSSION THREAD

**Discussion by ann0see** [jamulussoftware/jamulus#3082] [Ideas]
During the creation of the KB we prepared to add commenting abilities to the posts. I've now found a GitHub project which adds synchronization with GitHub discussions: https://giscus.app/ what do you think? Should we include this? Adding it seems quite easy.

**Comment by mcfnord** [comment]
Yes we should. I like retaining context and easy discovery of feedback and discussion options. We could always turn it off it isn't right.

**Reply by ann0see** [reply]
It needs access to the organisation or at least to this repository. This has security implications.

**Comment by ann0see** [comment]
@gilgongo what do you think about adding the ability to comment on posts from the KB via GitHub discussions?

**Reply by gilgongo** [reply]
As long as it doesn't promote the problem we've had in the past, which this is fragmentation of channels. At one point, we had Facebook, SourceForge, GH Issues, GH Discussions and GH Groups. We didn't have enough people to monitor/reply to all of them, so we reduced that to just two channels and managed to get unfocussed stuff out of GH Issues. 

So it might depend on the way it works. I wouldn't want to check/admin both GH Discussions and the KB for support requests, etc. if we could avoid that.

**Reply by ann0see** [reply]
No. The discussions will show up here.

**Comment by ann0see** [comment]
@gilgongo @mcfnord I've now enabled it by installing the giscus app on this repo, creating #3088 and doing some changes to the website. The live discussion on the site can be seen here: https://jamulus.io/kb/2022/01/26/Routing-JACK-Windows.html

I think the privacy page needs an update that we use giscus in the KB.

To add a discussion to a post:
1. Create a discussion in GH discussions
2. Copy the number of the discussion to the file in the _posts/...posttitle.md file as you can see here: https://github.com/jamulussoftware/jamuluswebsite/commit/22b9248f6eae8f6420b969fe53d32320fdce55f2

**Reply by mcfnord** [reply]
This looks really cool. I think the privacy implications of commenting are self-evident. I don't quite understand posttitle.md. The commit you show just adds a **discussionnumber** metadata item. I hope it's that easy!

**Reply by ann0see** [reply]
Ah sorry. posttitle.md is just a placeholder for the actual file containing the content of the KB entry. And yes, you just need to add discussionnumber.

The privacy implications are *not* self-evident - at least in the EU we should be cautious.

**Reply by mcfnord** [reply]
Just one discussion per post? Ideally, many could appear, but I bet just one can.

**Reply by ann0see** [reply]
I don't think multiple discussions can show up, but I don't think that's needed as there are threads like here which allow multiple discussions.

**Comment by mcfnord** [comment]
It doesn't look like there's any interactivity until a **discussionnumber** value has been added to the page. Now that the feature is on, is there value to connecting existing Discussions up to existing web pages where the discussion provides supplementary details? Or should we consider adding a blank discussion to every page in one step?

**Reply by ann0see** [reply]
I did both for testing. 

I think if there's a strong link to an existing discussion it's worth linking it. Otherwise just open a. We'd just open a new one and link it.
